### PR TITLE
[EOC-24] add react router and setup routing

### PR DESCRIPTION
**Pull request by alemikolo**

### CODE REVIEW

File: client/src/app/components/Layout/Layout.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        <Toolbar />
+        <Switch>
+          <Route path="/dashboard" component={Dashboard} />
+          <Route path="/cohort/:id(\w+)" component={Cohort} />
+          <Route path="/list/:id(\w+)" component={ShoppingList} />
+          <Route component={Dashboard} />
+        </Switch>
+      </Fragment>
+    );
+    return routes;
```
Comment (adamklepacz) [inline client/src/app/components/Layout/Layout.jsx]: do we need this if ternary operator is "automaticly" returning an expression?

```
return !currentUser ? (
      <Switch>
        <Route exact path="/" component={AuthBox} />
        <Redirect to="/" />
      </Switch>
    ) : (
      <Fragment>
        <Toolbar />
        <Switch>
          <Route path="/dashboard" component={Dashboard} />
          <Route path="/cohort/:id(\w+)" component={Cohort} />
          <Route path="/list/:id(\w+)" component={ShoppingList} />
          <Route component={Dashboard} />
        </Switch>
      </Fragment>
    );
```

File: client/src/app/components/Layout/Layout.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  currentUser: UserPropType,
+  history: PropTypes.shape({
+    push: PropTypes.func
+  }),
+
+  setCurrentUser: PropTypes.func.isRequired
+};
+
+const mapStateToProps = state => ({
+  currentUser: state.currentUser
```
Comment (adamklepacz) [inline client/src/app/components/Layout/Layout.jsx]: use selector here 
```
import {getCurrentUser} from 'selectors'

curentUser: getCurrentUser(state),
```

File: client/src/app/components/Layout/Layout.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    const routes = !currentUser ? (
+      <Switch>
+        <Route exact path="/" component={AuthBox} />
+        <Redirect to="/" />
+      </Switch>
+    ) : (
+      <Fragment>
+        <Toolbar />
+        <Switch>
+          <Route path="/dashboard" component={Dashboard} />
```
Comment (adamklepacz) [inline client/src/app/components/Layout/Layout.jsx]: props alphabetically

File: client/src/app/components/Layout/Layout.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  componentDidMount() {
+    this.setAuthenticationState();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { currentUser, history } = this.props;
+    const {
+      location: { pathname }
+    } = prevProps;
+    if (!prevProps.currentUser && currentUser) {
```
Comment (adamklepacz) [inline client/src/app/components/Layout/Layout.jsx]: please destructure this from prev props to make it more readable and understandable

File: client/src/app/components/Layout/Layout.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    this.setAuthenticationState();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { currentUser, history } = this.props;
+    const {
+      currentUser: prevCurrentUser,
+      location: { pathname }
+    } = prevProps;
+    if (!prevCurrentUser && currentUser) {
```
Comment (marekrozmus) [inline client/src/app/components/Layout/Layout.jsx]: add new line between consts and rest of code

File: client/src/app/components/Layout/Layout.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  setAuthenticationState = () => {
+    const { setCurrentUser } = this.props;
+    setCurrentUser();
+  };
+
+  render() {
+    const { currentUser } = this.props;
+    return !currentUser ? (
```
Comment (marekrozmus) [inline client/src/app/components/Layout/Layout.jsx]: add new line between consts and rest of code

File: client/src/modules/legacy/UserBar/actions.js [hunk 7]
```diff
@@ -1,17 +1,24 @@
...
 // Dispatchers
-export const logoutCurrentUser = () =>
+export const logoutCurrentUser = () => dispatch =>
   fetch(`${ENDPOINT_URL}/logout`, {
     method: 'POST',
     mode: 'no-cors',
     credentials: 'include'
   })
-    .then(() => window.location.replace(FRONTED_URL))
+    .then(() => {
```
Comment (marekrozmus) [inline client/src/modules/legacy/UserBar/actions.js]: remove braces

File: client/src/modules/legacy/UserBar/actions.js [hunk 8]
```diff
@@ -1,17 +1,24 @@
...
+      dispatch(logoutUser(null));
+    })
     .catch(err => {
```
Comment (marekrozmus) [inline client/src/modules/legacy/UserBar/actions.js]: are those braces needed here?

File: client/src/modules/legacy/reducers/currentUserReducer.js [hunk 9]
```diff
@@ -1,9 +1,12 @@
 import { SET_CURRENT_USER } from 'modules/legacy/mainActions';
+import { LOGOUT_USER } from 'modules/legacy/UserBar/actions';
 
 const user = (state = null, action) => {
   switch (action.type) {
     case SET_CURRENT_USER:
       return action.payload;
+    case LOGOUT_USER:
+      return action.payload;
```
Comment (marekrozmus) [inline client/src/modules/legacy/reducers/currentUserReducer.js]: shouldn't it be `return null;` ?

**Pull request merged**